lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1265,23 +1252,10 @@ internal class EvaluatingCompiler(
 
     private fun compileCastHelper(value: PartiqlAst.Expr, asType: PartiqlAst.Type, metas: MetaContainer): ThunkEnv {
         val expThunk = compileAstExpr(value)
-        val typedOpParameter = asType.toTypedOpParameter(customTypedOpParameters)
+        val typedOpParameter = asType.toTypedOpParameter(customTypedOpParameters, plugin)
         if (typedOpParameter.staticType is AnyType) {
             return expThunk
         }
-        if (
-            compileOptions.typedOpBehavior == TypedOpBehavior.HONOR_PARAMETERS &&
-            asType is PartiqlAst.Type.ScalarType &&
-            TYPE_ALIAS_TO_SCALAR_TYPE[asType.alias.text] === BuiltInScalarType.FLOAT &&
-            asType.parameters.isNotEmpty() // if precision of FLOAT is explicitly specified in the original query
-        ) {
-            err(
-                "FLOAT precision parameter is unsupported",
-                ErrorCode.SEMANTIC_FLOAT_PRECISION_UNSUPPORTED,
-                errorContextFrom(asType.metas),
-                internal = false
-            )
-        }
```

Comment:
This is validation of `FLOAT` type parameters, and we are moving it to the `FloatType` defined  legacy plugin package.